bot/university/university.py

```diff
@@ -0,0 +1,378 @@
+        been invoked, a help message for this command will be posted instead.
+
+        Args:
+            ctx (discord.ext.commands.Context): The context in which the command was called.
+        """
+        if ctx.invoked_subcommand is None:
+            await ctx.send_help(ctx.command)
+
+    @ufind.command(name='staff')
+    async def ufind_get_staff_data(self, ctx: commands.Context, *, search_term: str):
+        """Command Handler for the `ufind` subcommand `staff`.
+
+        Allows users to search the service `ufind` for information regarding staff members and posts them on Discord.
+        If the search result contains more than one person, an embed with reactions will be posted. The user can then
+        choose a person by reacting to said embed. If the result contains no person at all, the bot will inform the
+        user by posting a sad message.
+
+        Args:
+            ctx (discord.ext.commands.Context): The context in which the command was called.
+            search_term (str): The term to be searched for (most likely firstname, lastname or both).
+        """
+        search_filters = "%20%2Be%20c%3A6"  # URL Encoding
+        response = requests.get(constants.URL_UFIND_API + "/staff/?query=" + search_term + search_filters)
+        response.encoding = "utf-8"
+
+        if response.status_code == 200:
+            xml = ET.fromstring(response.text)
+            persons = xml.findall("person")
+            index = 0
+
+            if len(persons) > 0:
+                if len(persons) > 1:
+                    index = await self._staff_selection(ctx.author, ctx.channel, persons)
+
+                response = requests.get(constants.URL_UFIND_API + "/staff/" + persons[index].attrib["id"])
+                response.encoding = "utf-8"
+
+                staff_data = _parse_staff_xml(response.text)
+                embed = _create_embed_staff(staff_data)
+                await ctx.channel.send(embed=embed)
+            else:
+                raise ValueError("No person with the specified name was found.")
+        else:
+            response.raise_for_status()
+
+    @ufind_get_staff_data.error
+    async def ufind_error(self, ctx, error: commands.CommandError):
+        """Error Handler for the `ufind` subcommand `staff`.
+
+        Handles specific exceptions which occur during the execution of this command. Other exceptions will be simply
+        re-raised so that they can be handled somewhere else.
+
+        Args:
+            ctx (discord.ext.commands.Context): The context in which the command was called.
+            error (commands.CommandError): The error raised during the execution of the command.
+        """
+        if isinstance(error.original, ValueError):
+            await ctx.send("Ich konnte leider niemanden unter dem von dir angegeben Namen finden. :slight_frown: "
+                           "Möglicherweise hast du dich verschrieben.")
+        else:
+            raise error
```

Comment:
Should we return a generic error msg to the user here?

Reply:
No. This exception here is raised again so that the global exception handler can catch and handle it. The local one should just handle specific ones.

Reply:
If you need to make a is Instance check to handle a single type of error,you should handle it in the command using except ValueError.

In my **opinion** a error Handler for a command does not make much sense for readability if we only handle one error and bubble the rest up anyway.

Reply:
There's only one error type for now. I'm gonna add additional subcommands in the future.